malpolon/models/standard_prediction_systems.py

```diff
@@ -448,4 +456,92 @@ def __init__(
                              'kwargs': {}}
             }
 
+        super().__init__(model, loss, optimizer, metrics)
+
+
+class RegressionSystem(GenericPredictionSystem):
+    """Regression task class."""
+    def __init__(
+        self,
+        model: Union[torch.nn.Module, Mapping],
+        lr: float = 1e-2,
+        weight_decay: float = 0,
+        momentum: float = 0.9,
+        nesterov: bool = True,
+        metrics: Optional[dict[str, Callable]] = None,
+        task: str = 'regression',
+        hparams_preprocess: bool = True,
+    ):
+        """Class constructor.
+
+        Parameters
+        ----------
+        model : dict
+            model to use
+        lr : float
+            learning rate
+        weight_decay : float
+            weight decay
+        momentum : float
+            value of momentum
+        nesterov : bool
+            if True, uses Nesterov's momentum
+        metrics : dict
+            dictionnary containing the metrics to compute.
+            Keys must match metrics' names and have a subkey with each
+            metric's functional methods as value. This subkey is either
+            created from the `malpolon.models.utils.FMETRICS_CALLABLES`
+            constant or supplied, by the user directly.
+        task : str, optional
+            Machine learning task (used to format labels accordingly),
+            by default 'regression'. The value determines
+            the loss to be selected.
+        hparams_preprocess : bool, optional
+            if True performs preprocessing operations on the hyperparameters,
+            by default True
+        """
+        if hparams_preprocess:
+            assert task == 'regression', "Regression task must be specified."
+            metrics = check_metric(metrics)
+
+
+        self.lr = lr
+        self.weight_decay = weight_decay
+        self.momentum = momentum
+        self.nesterov = nesterov
+
+        model = check_model(model)
+
+        optimizer = torch.optim.AdamW(model.parameters(),
+                                      lr=lr,
+                                      weight_decay=weight_decay)
+
+        print(optimizer)
+
+        if 'regression' in task:
+            loss = torch.nn.MSELoss()
+
+        if metrics is None:
+            metrics = {
+                "regression_R2score": {'callable': Fmetrics.regression.r2_score,
+                                       'kwargs': {}}
+            }
+
         super().__init__(model, loss, optimizer, metrics=metrics)
+
+    def configure_optimizers(self):
```

Comment:
INFO: since the v2.1.0 update, this method doesn't need to be re-defined unless you want to impose a fixed optimizer/scheduler as both of these objects can be defined by the user in his experiment config file

Reply:
withdrawn